_drafts/testing-third-party-apis.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: ¿Cómo probamos las APIs de terceros?
+author: arturopuente
+tags:
+  - rails
+  - testing
+  - api
+  - third-party
+---
+
+Un problema común que nos encontramos al testear nuestras aplicaciones es la integración con servicios de terceros.
+
+Enviar los requests a las APIs de terceros en cada test tiene varios problemas:
+
+- Nos topamos con el límite de requests en un determinado tiempo
+- Los tests pueden fallar por problemas de conexión
+- La suite de pruebas demora **mucho**
+
+Hay gemas que nos ayudan que nuestras pruebas funcionen mejor, por ejemplo:
+
+- **[Webmock](https://github.com/bblimke/webmock)** desactiva los requests a cualquier servicio externo e ir devolviendo respuestas predeterminadas por cada URL.
+
+- **[VCR](https://github.com/vcr/vcr)** ejecuta nuestros tests una vez contra los servicios reales y guarda las respuestas en un JSON (cassettes), de forma que la siguiente vez que la suite se ejecute ese JSON servirá para mockear la respuesta del servidor.
+
+Sin embargo, todas estas gemas añaden una capa de complejidad a nuestra suite:
+
+- Con Webmock si queremos que hacer requests a un servicio de terceros en una prueba tenemos que manejar manualmente la activación/desactivación de los requests externos
+
+- Al usar VCR tenemos que manejar los cassettes y como contienen toda la interacción entre los servidores, manipularlos y editarlos es más complejo y tedioso.
+
+¿Qué alternativa tenemos entonces? Me parece que la respuesta está en usar RSpec hasta que la complejidad de nuestra suite nos obligue a buscar una gema adicional, así evitamos añadir una dependencia más y evitamos los problemas/workarounds que las otras gemas traen consigo.
+
+## Usando RSpec para mockear servicios externos
```

Comment:
No había escuchado nunca ese término pero me suena a la españolización del verbo inglés `mock`: `mock the service to...`, que es bastante usado en la literatura de tests. Quizás una buena traducción sería `simular`.